modules/dnn/src/dnn.cpp

```diff
@@ -2210,7 +2238,17 @@ void Net::setInput(InputArray blob, const String& name)
     ld.outputBlobs.resize( std::max(pin.oid+1, (int)ld.requiredOutputs.size()) );
     ld.outputBlobsWrappers.resize(ld.outputBlobs.size());
     MatShape prevShape = shape(ld.outputBlobs[pin.oid]);
-    Mat blob_ = blob.getMat();
+    Mat blob_;
+    if (impl->preferableTarget == DNN_TARGET_OPENCL &&
+        impl->preferablePrecision == DNN_PRECISION_FP16)
```

Comment:
I think we can use just [DNN_TARGET_OPENCL_FP16](https://docs.opencv.org/master/d6/d0f/group__dnn.html#gga709af7692ba29788182cf573531b0ff5ab6d2643e6e5a3d0dfaca2ec69fd041b6) target instead of separate precisions enum.

Reply:
yes, this flag should be good. patch is updated.